Src/EB/AMReX_EB_triGeomOps_K.H

```diff
@@ -0,0 +1,303 @@
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE Real side_op(Real L1[6],Real L2[6])
+    {
+        return(    L1[0]*L2[4] 
+                +  L1[1]*L2[5] 
+                +  L1[2]*L2[3] 
+                +  L1[3]*L2[2] 
+                +  L1[4]*L2[0] 
+                +  L1[5]*L2[1] );
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void getvec(Real P1[3],Real P2[3],Real v[3])
+    {
+        v[0]=P2[0]-P1[0];
+        v[1]=P2[1]-P1[1];
+        v[2]=P2[2]-P1[2];
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void getunitvec(Real v[3],Real vu[3])
+    {
+        Real vmag;
+        vmag=sqrt(v[0]*v[0]+v[1]*v[1]+v[2]*v[2]);
+        vu[0]=v[0]/vmag;
+        vu[1]=v[1]/vmag;
+        vu[2]=v[2]/vmag;
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void CrossProd(Real v1[3],Real v2[3],Real v[3])
+    {
+        v[0]=v1[1]*v2[2]-v1[2]*v2[1];
+        v[1]=v1[2]*v2[0]-v1[0]*v2[2];
+        v[2]=v1[0]*v2[1]-v1[1]*v2[0];
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void get_plucker_coords(Real v1[3],Real v2[3],Real L[6])
+    {
+        L[0] = v1[0]*v2[1] - v1[1]*v2[0];
+        L[1] = v1[0]*v2[2] - v1[2]*v2[0];
+        L[2] = v1[0]       - v2[0];
+        L[3] = v1[1]*v2[2] - v1[2]*v2[1];
+        L[4] = v1[2]       - v2[2];
+        L[5] = v2[1]       - v1[1];
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void side_op3(Real v1[3],Real v2[3],
+            Real t1[3],Real t2[3],Real t3[3],
+            Real &S1, Real &S2, Real &S3)
+    {
+
+        Real L[6],e1[6],e2[6],e3[6];
+
+        get_plucker_coords(v1,v2,L);
+        get_plucker_coords(t1,t2,e1);
+        get_plucker_coords(t2,t3,e2);
+        get_plucker_coords(t3,t1,e3);
+
+        S1=side_op(L,e1);
+        S2=side_op(L,e2);
+        S3=side_op(L,e3);
+    }
+    //================================================================================
+    //get normal of triangle pointing at a test-point
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void tri_n(Real P1[3],Real P2[3],Real P3[3],
+            Real testp[3],Real n[3])
+    {
+        Real v1[3],v2[3],magn;
+        Real centr[3],c_tp_vec[3];
+
+        getvec(P1,P2,v1);
+        getvec(P1,P3,v2);
+        CrossProd(v1,v2,n);
+
+
+        centr[0]=0.333333*(P1[0]+P2[0]+P3[0]);
+        centr[1]=0.333333*(P1[1]+P2[1]+P3[1]);
+        centr[2]=0.333333*(P1[2]+P2[2]+P3[2]);
+
+        getvec(centr,testp,c_tp_vec);
+        magn=sqrt(n[0]*n[0]+n[1]*n[1]+n[2]*n[2]);
+
+        if(DotProd(c_tp_vec,n) < 0.0)
```

Comment:
Can we make the constants in the PR something like `Real(0.0)` so that the compile does not complain for single precision builds?

Reply:
Actually, don't worry about this.  The EB codes won't work with single precision anyway because there are so many other places that need to be fixed.

Reply:
not worrying about this now!